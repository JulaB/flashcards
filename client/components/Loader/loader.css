@keyframes dot-bounce {
  0%,
  100% { transform: scale(0); }
  50% { transform: scale(1); }
}

.loader {
  display: flex;
  margin: 0 auto;
  width: 66px;

  &__dot1,
  &__dot2,
  &__dot3 {
    width: 16px;
    height: 16px;
    margin: 3px;
    border-radius: 50%;
    background-color: var(--color-loader);
    animation: dot-bounce 1.4s ease-in-out infinite;
  }

  &__dot2 {
    animation-delay: 0.16s;
  }

  &__dot3 {
    animation-delay: 0.32s;
  }

  &--btn {
    width: auto;

    .loader__dot1,
    .loader__dot2,
    .loader__dot3 {
      width: calc(var(--text-base-size) - 6px);
      height: calc(var(--text-base-size) - 6px);
      background-color: var(--btn-primary-text);
    }
  }
}
