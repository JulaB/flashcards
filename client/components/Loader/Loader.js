import React from 'react';
import PropTypes from 'prop-types';
import './loader.css';

const Loader = ({ className }) => (
  <div className={`loader ${className}`}>
    <div className="loader__dot1" />
    <div className="loader__dot2" />
    <div className="loader__dot3" />
  </div>
);

Loader.propTypes = {
  className: PropTypes.string,
};

Loader.defaultProps = {
  className: '',
};

export default Loader;
