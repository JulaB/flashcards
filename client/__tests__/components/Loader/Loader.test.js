import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loader from 'components/Loader/Loader';

describe('<Loader />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<Loader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
