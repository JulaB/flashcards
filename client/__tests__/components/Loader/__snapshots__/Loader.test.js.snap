// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders the component 1`] = `
<div
  className="loader "
>
  <div
    className="loader__dot1"
  />
  <div
    className="loader__dot2"
  />
  <div
    className="loader__dot3"
  />
</div>
`;
